File: docs/breez-sdk/snippets/python/src/turnkey.py
from breez_sdk_spark import (
    default_config,
    create_turnkey_signer,
    connect_with_signer,
    BreezSdk,
    ConnectWithSignerRequest,
    Network,
    TurnkeyConfig,
)

async def connect_with_turnkey() -> BreezSdk:
    # ANCHOR: turnkey-connect
    turnkey_config = TurnkeyConfig(
        base_url=None,
        organization_id="<turnkey sub-organization id>",
        api_public_key="<api public key hex>",
        api_private_key="<api private key hex>",
        wallet_id="<turnkey wallet id>",
        network=Network.MAINNET,
        account_number=None,
        # Set after the first connect to make later signer setup network-free
        identity_public_key=None,
        retry=None,
        max_rps=None,
    )

    signers = await create_turnkey_signer(config=turnkey_config)

    config = default_config(Network.MAINNET)
    config.api_key = "<breez api key>"

    sdk = await connect_with_signer(ConnectWithSignerRequest(
        config=config,
        breez_signer=signers.breez_signer,
        spark_signer=signers.spark_signer,
        storage_dir="./.data"
    ))
    # ANCHOR_END: turnkey-connect
    return sdk

File: docs/breez-sdk/snippets/rust/src/turnkey.rs
use anyhow::Result;
use breez_sdk_spark::turnkey::{create_turnkey_signer, TurnkeyConfig};
use breez_sdk_spark::*;

async fn connect_with_turnkey() -> Result<BreezSdk> {
    // ANCHOR: turnkey-connect
    let turnkey_config = TurnkeyConfig {
        base_url: None,
        organization_id: "<turnkey sub-organization id>".to_string(),
        api_public_key: "<api public key hex>".to_string(),
        api_private_key: "<api private key hex>".to_string(),
        wallet_id: "<turnkey wallet id>".to_string(),
        network: Network::Mainnet,
        account_number: None,
        // Set after the first connect to make later signer setup network-free
        identity_public_key: None,
        retry: None,
        max_rps: None,
    };

    let signers = create_turnkey_signer(turnkey_config).await?;

    let mut config = default_config(Network::Mainnet);
    config.api_key = Some("<breez api key>".to_string());

    let sdk = connect_with_signer(ConnectWithSignerRequest {
        config,
        breez_signer: signers.breez_signer,
        spark_signer: signers.spark_signer,
        storage_dir: "./.data".to_string(),
    })
    .await?;
    // ANCHOR_END: turnkey-connect
    Ok(sdk)
}
